proto/UiOptions.proto

```diff
@@ -30,7 +30,7 @@ message TextField {
     string text = 1;
 }
 
-message UiOption {
+message UiOptionDeclaration {
```

Comment:
Looks good, the one thing I'm unsure about is how we'll toggle the mutability of a field. Should this be a boolean here?

Reply:
Good one, I'll add it. Worst case we don't end up using/needing it.

Reply:
I also forgot to mention field descriptions we discussed - it appears those are missing as well.